FWI_forward_adjoint/FWIpart1.md

```diff
@@ -56,65 +56,68 @@ create highly optimized low-level C code. For this purpose Devito uses
-  objects via shorthand expressions, enabling the syntax `f.dx` to
-  denote $\frac{\partial f}{\partial x}$ and `f.dx2` for
-  $\frac{\partial^2 f}{\partial x^2}$.
+  objects via shorthand expressions, where the `space_order` parameter
+  defines the order of the created finite difference stencil.
 
-* `TimeData` objects represent a time-dependent function
+* `TimeFunction` objects represent a time-dependent function
   that includes leading dimension $t$, for example `g(t, x, y)`. In
-  addition to spatial derivatives `TimeData` symbols also provide
-  time derivatives `g.dt` and `g.dt2`, as well as options to save
+  addition to spatial derivatives `TimeFunction` symbols also provide
+  time derivatives `g.dt` and `g.dt2`, as well as options to store
   the entire data along the time axis.
 
 To demonstrate Devito's symbolic capabilities, let us consider a
 time-dependent function $\vd{u}(t, x, y)$ representing the discrete forward
 wavefield. We can define this as a `TimeData` object in Devito:
 
 ```python
-    u = TimeData(name="u", shape=model.shape_domain, time_order=2,
-                 space_order=2, save=True, time_dim=nt)
+  u = TimeData(name='u', grid=model.grid, time_order=2,
+               space_order=2, save=True, timd_dim=nt)
 ```
 
-where the parameter `shape` defines the size of the allocated memory
-region, `time_order` and `space_order` define the default
-discretization order of the derivative expressions and the parameters
-`save=True` and `time_dim` force the entire wavefield to be stored in
-memory.
+where the the `grid` object provided by the `model` defines the size
+of the allocated memory region, `time_order` and `space_order` define
+the default discretization order of any derived derivative expressions.
 
-We can now use this wavefield to generate simple discretized stencil
-expressions for finite difference derivatives as:
+We can now use this symbilic representation of our wavefield to
+generate simple discretized stencil expressions for finite difference
+derivatives using shorthand expressions, such as `u.dx` and `u.dx2` to
+denote $\frac{\partial f}{\partial x}$ and $\frac{\partial^2
+f}{\partial x^2}$ respectively.
 
 ```python
   In []: u
   Out[]: u(t, x, y)
 
-  In []: u.dt
-  Out[]: -u(t, x, y)/s + u(t+s, x, y)/s
+  In []: u.dx
+  Out[]: -u(t, x, y)/h_x + u(t, x + h_x, y)/h_x
 
-  In []: u.dt2
-  Out[]: -2*u(t, x, y)/s**2 + u(t-s, x, y)/s**2 + u(t+s, x, y)/s**2
+  In []: u.dx2
+  Out[]: -2*u(t, x, y)/h_x**2 + u(t, x-h_x, y)/h_x**2 + u(t, x-h_x, y)/h_x**2
 ```
 
 Using the automatic derivation of derivative expressions we can now
 implement a discretized expression for Equation #WE\ without the
 source term $q(x,y,t;x_s, y_s)$. The `model` object which we created
 earlier, already contains the squared slowness $\vd{m}(x, y)$ and 
-damping term $\vd{\eta}(x, y)$ as `DenseData` objects:
+damping term $\vd{\eta}(x, y)$ as `Function` objects:
```

Comment:
no (x,y), \vd{\eta} is discrete and implicit indices notation.

Reply:
Yeah, the switch to discrete notation in this section is extremely confusing, IMHO. I thought the idea was to stay close to the actual SymPy code and print out the Latex SymPy generates for the final stencil? Switching to discrete notation does not really help makes this easier to follow, afaict.

Reply:
It does for TLE audience that always write it like and and follow that convention. Not saying it's right, just that it's the way geophys like it.